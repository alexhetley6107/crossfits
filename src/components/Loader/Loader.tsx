import React from 'react';
import styled from 'styled-components';
import { Icon } from '../Icon';

export type LoaderProps = {
  className?: string;
  size?: number;
  color?: string;
};

const LoaderWrapper = styled.div`
  & svg {
    animation: spin 1s linear infinite;
  }

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export const Loader: React.FC<LoaderProps> = ({ size = 20, color, className }) => {
  return (
    <LoaderWrapper data-testid="Loader" className={className} style={{ height: size + 'px' }}>
      <Icon name="Load" size={size} color={color} />
    </LoaderWrapper>
  );
};
