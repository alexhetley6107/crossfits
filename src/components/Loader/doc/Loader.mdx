import { Canvas, Meta } from '@storybook/blocks';
import { Example } from './Example.tsx';
import * as Loader from './Loader.stories.tsx';

<Meta of={Loader} title="Loader" />

# Loader

<Canvas of={Loader.Base} />

## Usage

```ts
import { Loader } from 'remindee-ui-kit';

export const Example = () => {
  return <Loader />;
};
```

### Arguments

- **className** - A string that sets css class to the root of component.