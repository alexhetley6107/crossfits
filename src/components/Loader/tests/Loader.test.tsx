import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Loader } from '../Loader';
import '@testing-library/jest-dom';
import 'jest-styled-components';

describe('Loader component', () => {
  describe('General', () => {
    it('Render', () => {
      render(<Loader />);
      const el = screen.getByTestId('Loader');
      expect(el).toBeInTheDocument();
    });
  });
  describe('Props', () => {
    it('Color', () => {
      render(<Loader color="red" />);
      const el = screen.getByTestId('Loader');
      expect(el).toMatchSnapshot();
    });
    it('Size', () => {
      render(<Loader size={30} />);
      const el = screen.getByTestId('Loader');
      expect(el).toMatchSnapshot();
    });
  });
});
