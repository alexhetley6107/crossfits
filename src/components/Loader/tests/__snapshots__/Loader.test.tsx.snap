// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader component > Props > Color 1`] = `
.c0 svg {
  animation: spin 1s linear infinite;
}

<div
  class="c0"
  data-testid="Loader"
  style="height: 20px;"
>
  <svg
    color="red"
    data-testid="Icon"
    fill="none"
    height="20"
    width="20"
  >
    <use
      href="/sprite.svg#Load"
    />
  </svg>
</div>
`;

exports[`Loader component > Props > Size 1`] = `
.c0 svg {
  animation: spin 1s linear infinite;
}

<div
  class="c0"
  data-testid="Loader"
  style="height: 30px;"
>
  <svg
    data-testid="Icon"
    fill="none"
    height="30"
    width="30"
  >
    <use
      href="/sprite.svg#Load"
    />
  </svg>
</div>
`;
